numpy/core/src/multiarray/descriptor.c

```diff
@@ -1711,36 +1687,32 @@ _convert_from_bytes(PyObject *obj, PyArray_Descr **at)
                 }
             }
         }
-        return PyArray_DescrConverter(item, at);
+        return _convert_from_any(item);
     }
 
-    if (PyDataType_ISUNSIZED(*at) && (*at)->elsize != elsize) {
-        PyArray_DESCR_REPLACE(*at);
-        if (*at == NULL) {
+    if (PyDataType_ISUNSIZED(ret) && ret->elsize != elsize) {
+        PyArray_DESCR_REPLACE(ret);
+        if (ret == NULL) {
             return NPY_FAIL;
```

Comment:
```suggestion
            return NULL;
```

Reply:
There is something weird with the whitespace rendering (rendering a deleted space in front), so I am not sure if these applies correctly or not.

Reply:
I've seen that behavior before, and had it bite me - but in this case, it did apply correctly